frontends/verilog/verilog_parser.y

```diff
@@ -1447,10 +1530,10 @@ enum_base_type: type_atom type_signing
 	| %empty			{ astbuf1->is_reg = true; addRange(astbuf1); }
 	;
 
-type_atom: TOK_INTEGER		{ astbuf1->is_reg = true; addRange(astbuf1); }		// 4-state signed
-	|  TOK_INT		{ astbuf1->is_reg = true; addRange(astbuf1); }		// 2-state signed
-	|  TOK_SHORTINT		{ astbuf1->is_reg = true; addRange(astbuf1, 15, 0); }	// 2-state signed
-	|  TOK_BYTE		{ astbuf1->is_reg = true; addRange(astbuf1,  7, 0); }	// 2-state signed
+type_atom: TOK_INTEGER		{ astbuf1->is_reg = true; astbuf1->is_signed = true; addRange(astbuf1); }		// 4-state signed
+	|  TOK_INT		{ astbuf1->is_reg = true; astbuf1->is_signed = true; addRange(astbuf1); }		// 2-state signed
+	|  TOK_SHORTINT		{ astbuf1->is_reg = true; astbuf1->is_signed = true; addRange(astbuf1, 15, 0); }	// 2-state signed
+	|  TOK_BYTE		{ astbuf1->is_reg = true; astbuf1->is_signed = true; addRange(astbuf1,  7, 0); }	// 2-state signed
```

Comment:
It is related, see test: https://github.com/YosysHQ/yosys/pull/2248/commits/9f1cbefc6cd8e04338d3cf4c97f203fc02c8e438#diff-f314296647b6b3ec5688a429d97fa6c2a39a48ff9aa95edfe1b7dadb8b6ed5a0R15

This PR adds possibility to declare loop iterator as a part of the loop declaration and this variable can be e.g. ``int``. Without this change, loops that are checking, if variable is ``>= 0`` never ends (as ``int`` there is marked as unsigned).

Reply:
`type_atom` is only used for structs and enums, which doesn't apply to any of the test cases included in this PR. I don't doubt that there is a bug here, but I believe it would be prudent to fix that bug in a separate PR with appropriate test coverage.

Reply:
I moved this change to: https://github.com/YosysHQ/yosys/pull/2584

You are right, I was using ``member_type`` in ``local_definition_stmt`` so I thought this change is also needed here. Now I'm using ``non_io_wire_type`` instead.